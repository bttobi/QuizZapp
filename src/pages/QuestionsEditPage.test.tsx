import { Route, Routes } from 'react-router-dom';
import { QuestionsEditPage } from '.';
import { render, waitFor } from '../tests/test-utils';

const setup = () =>
  render(
    <Routes>
      <Route
        path="/questions/edit/:quizID"
        element={<QuestionsEditPage />}
      ></Route>
    </Routes>,
    { initialEntries: ['/questions/edit/1'] }
  );

describe('QuestionsEditPage component', () => {
  it('renders the component', async () => {
    const { container } = setup();
    await waitFor(() => {
      expect(container).toBeInTheDocument();
    });
  });

  it('displays fetched questions', async () => {
    const { getAllByRole, getByText } = setup();
    await waitFor(() => {
      expect(getAllByRole('button')).toHaveLength(5);
      expect(getByText('Question')).toBeInTheDocument();
      expect(getByText('Select all questions')).toBeInTheDocument();
      expect(getByText('Delete')).toBeInTheDocument();
      expect(getByText('Currently editing:')).toBeInTheDocument();
      expect(getByText('My test')).toBeInTheDocument();
      expect(getByText('Test?')).toBeInTheDocument();
    });
  });

  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
