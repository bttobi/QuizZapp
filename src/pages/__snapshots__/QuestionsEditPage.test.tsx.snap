// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuestionsEditPage component > matches snapshot 1`] = `
<div>
  <div
    class="w-full h-full flex flex-col justify-start align-center items-center"
    style="opacity: 0;"
  >
    <div
      class="text-3xl mt-4 flex flex-col justify-center items-center align-center"
    >
      <span
        class="text-center"
      >
        Currently editing: 
      </span>
      <span
        class="text-center mt-4"
      >
        My test
      </span>
    </div>
    <section
      class="flex flex-col gap-4 mt-4 items-center align-center justify-center"
    >
      <div
        class="group flex flex-col w-full max-w-xs"
        data-has-elements="true"
        data-has-label="true"
        data-has-value="true"
        data-slot="base"
      >
        <div
          class="relative w-full inline-flex tap-highlight-transparent shadow-sm px-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-unit-10 rounded-medium flex-col items-start justify-center gap-0 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background h-14 py-2"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <label
            class="absolute z-10 pointer-events-none origin-top-left subpixel-antialiased block text-foreground-500 cursor-text will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-default-600 group-data-[filled-within=true]:pointer-events-auto group-data-[filled-within=true]:scale-85 text-small group-data-[filled-within=true]:-translate-y-[calc(50%_+_theme(fontSize.small)/2_-_6px)] pe-2 max-w-full text-ellipsis overflow-hidden"
            data-slot="label"
            for="react-aria-:r13:"
            id="react-aria-:r14:"
          >
            Question
          </label>
          <div
            class="inline-flex w-full items-center h-full box-border group-data-[has-label=true]:items-end pb-0.5"
            data-slot="inner-wrapper"
          >
            <input
              aria-label="Question"
              aria-labelledby="react-aria-:r13: react-aria-:r14:"
              class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 text-small peer pr-6 group-data-[has-value=true]:text-default-foreground"
              data-slot="input"
              id="react-aria-:r13:"
              type="text"
              value=""
            />
            <span
              class="p-2 -m-2 z-10 hidden absolute right-3 appearance-none select-none opacity-0 hover:!opacity-100 cursor-pointer active:!opacity-70 rounded-full outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-large peer-data-[filled=true]:opacity-70 peer-data-[filled=true]:block transition-opacity motion-reduce:transition-none"
              data-slot="clear-button"
              role="button"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="1em"
                role="presentation"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <label
        class="group relative max-w-fit cursor-pointer tap-highlight-transparent p-2 -m-2 bg-default-50 rounded-xl flex justify-center align-center items-center"
      >
        <div
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-label=" "
            aria-labelledby=":r18:"
            type="checkbox"
            value=""
          />
        </div>
        <span
          aria-hidden="true"
          class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:bg-primary after:text-primary-foreground text-primary-foreground w-6 h-6 mr-2 rounded-[calc(theme(borderRadius.medium)*0.7)] before:rounded-[calc(theme(borderRadius.medium)*0.7)] after:rounded-[calc(theme(borderRadius.medium)*0.7)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none"
        >
          <svg
            aria-hidden="true"
            class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-5 h-4 transition-opacity motion-reduce:transition-none"
            role="presentation"
            viewBox="0 0 17 18"
          >
            <polyline
              fill="none"
              points="1 9 7 14 15 4"
              stroke="currentColor"
              stroke-dasharray="22"
              stroke-dashoffset="66"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
        <span
          class="relative text-foreground select-none text-large transition-colors-opacity before:transition-width motion-reduce:transition-none"
          id=":r18:"
        >
          <span
            class="text-sm"
          >
            Select all questions
          </span>
        </span>
      </label>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-danger data-[hover=true]:opacity-hover text-white w-48"
        data-disabled="true"
        disabled=""
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          />
        </svg>
        Delete
      </button>
    </section>
    <ul
      class="grid grid-col-1 justify-center items-center align-center gap-4 mt-12 p-4 bg-backgroundSecondary rounded-lg mb-32 mx-4"
    >
      <li
        style="transform: scale(0) translateZ(0);"
      >
        <div
          class="flex cursor-pointer justify-center items-center align-center bg-default p-4 rounded-xl border-3 border-primary flex-grow"
        >
          <label
            class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 -m-2 z-10 bg-primary rounded-lg w-10"
          >
            <div
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                aria-label=" "
                aria-labelledby=":r1h:"
                type="checkbox"
                value=""
              />
            </div>
            <span
              aria-hidden="true"
              class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:bg-primary after:text-primary-foreground text-primary-foreground w-6 h-6 mr-2 rounded-[calc(theme(borderRadius.medium)*0.7)] before:rounded-[calc(theme(borderRadius.medium)*0.7)] after:rounded-[calc(theme(borderRadius.medium)*0.7)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none"
            >
              <svg
                aria-hidden="true"
                class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-5 h-4 transition-opacity motion-reduce:transition-none"
                role="presentation"
                viewBox="0 0 17 18"
              >
                <polyline
                  fill="none"
                  points="1 9 7 14 15 4"
                  stroke="currentColor"
                  stroke-dasharray="22"
                  stroke-dashoffset="66"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
          </label>
          <span
            class="mx-5"
            style="word-break: break-word; hyphens: auto;"
          >
            Test?
          </span>
          <div
            class="ml-auto flex gap-2"
          >
            <button
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-unit-2 rounded-medium px-unit-0 !gap-unit-0 data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-secondary text-secondary-foreground min-w-unit-10 w-unit-10 h-unit-10 data-[hover=true]:opacity-hover"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                />
              </svg>
            </button>
            <button
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-unit-2 rounded-medium px-unit-0 !gap-unit-0 data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-danger text-danger-foreground min-w-unit-10 w-unit-10 h-unit-10 data-[hover=true]:opacity-hover"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
    </ul>
    <button
      class="z-0 group inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-unit-3 px-unit-0 !gap-unit-0 data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-success min-w-unit-12 data-[hover=true]:opacity-hover fixed bottom-12 right-12 rounded-full text-white text-3xl w-16 h-16"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </button>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
